src/common/tse.c

```diff
@@ -269,6 +269,20 @@ tse_task_decref(tse_task_t *task)
 	D_FREE(task);
 }
 
+static void
+tse_task_decref_free_locked(tse_task_t *task)
```

Comment:
or we can just add another parameter to tse_task_decref_locked(..., bool free_on_zero)